### PR TITLE
Update ``primer`` tests to own file, separate batches and concurrent runs

### PR DESCRIPTION
**Pull request by DanielNoord**
- [x] Add yourself to CONTRIBUTORS if you are a new contributor.
- [x] Write a good description on what the PR does.

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :sparkles: New feature |
| ✓   | :hammer: Refactoring   |
| ✓   | :scroll: Docs          |

## Description

First set of improvements for the `primer`. This adds:
- Concurrent jobs, meaning the jobs will stop if a new action is created.
- `primer` now only runs on changes to `pylint/**` or `tests/primer/**` or pushes to `main`. This should allow updating docs or tests without the primer holding you up.
- The `primer` sits in its own workflow file
- The `primer` has been bursted into two batches based on the first half of the alphabet and the second 😄 

Ref #5359

### CODE REVIEW

**Comment by DanielNoord** [comment]
Times between both batches seem fairly equal so I think this division works. Run time is quite high as US is starting to launch GitHub Actions at this time..

Looks good for review!

**Review by Pierre-Sassoulas — COMMENTED** [review]
I just had a quick look but it look amazing already !I'll do a full review tomorrow.

I was reading the doc for github caching so we an download the repositories once. I think if we choose a particular commit we can even cache reliably between runs as long as we do not update to the latest commits. I'll do that in another MR though, everything here was required, is compatible and will also make things anyway !

File: .github/workflows/primer-test.yaml [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+        run: |
+          echo "Failed to restore Python venv from cache"
+          exit 1
+      - name: Run pytest
+        run: |
+          . venv/bin/activate
+          pip install -e .
+          pytest -m primer_external_batch_one --primer-external -n auto
+
+  pytest-primer-external-batch-two:
```
Comment (Pierre-Sassoulas) [inline .github/workflows/primer-test.yaml:133]: I was trying to find a way to not copy paste so much, did you also search and find nothing acceptable or did you simply implement something fast ?

**Review by Pierre-Sassoulas — APPROVED** [review]
I think we can merge as is, but I have a small question.

File: .github/workflows/primer-test.yaml (hunk 1 above)
Comment (DanielNoord) [reply .github/workflows/primer-test.yaml:133]: I simply implemented something fast. Do you want me to look if something is possible before merging?

Comment (Pierre-Sassoulas) [reply .github/workflows/primer-test.yaml:133]: No, let's keep the MR small and do it later on. Also this CI speed up is too valuable to wait, clearly 😄

**Pull request merged**